File: C_Programming/Practise_Programs/PP_162.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_162
//
//  Description       : This program demonstrates the left shift
//                      operation on an unsigned integer by displaying
//                      hexadecimal values for bit positions from
//                      1 to 33.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 20/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UNIT;

int main()
{
    UNIT iMask = 1;
    int iCnt = 0;

    for(iCnt = 1; iCnt <= 33; iCnt++)
    {
        printf("%d : %X\n", iCnt, iMask);
        iMask = iMask << 1;
    }

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      1 : 1
//      2 : 2
//      3 : 4
//      4 : 8
//      5 : 10
//      ...
//      32 : 80000000
//      33 : 0
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_162.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_162
//
//  Description       : This program accepts a number from the
//                      user and displays a square pattern of
//                      stars. The number of rows and columns
//                      is determined by the given number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 04/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_162
{
    public static void Display(int iNo)
    {
        int i , j = 0;

        for(i = 1; i <= iNo; i++)
        {
            for(j = 1; j <= iNo; j++)
            {
                System.out.print("*\t");
            }
            System.out.println();
        }
        
    }
    public static void main(String A[])
    {        
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the elements : 5
//
//  Sample Output :
//
//      *	*	*	*	*
//      *	*	*	*	*
//      *	*	*	*	*
//      *	*	*	*	*
//      *	*	*	*	*
//
//////////////////////////////////////////////////////////////////